sshconfig: add indent file

jaki opened the pull request:
Add indent file for `sshconfig` filetype.  Have it indent "Host" and
"Match" lines at indent level 0 and lines under those sections at indent
level 1.

jaki committed:
feat(indent): add sshconfig

Add indent file for `sshconfig` filetype.  Handle indenting items under
`Host` and `Match`.  Make sure to handle it case-insensitive.  Make sure
to handle all whitespace, not just spaces.  Make sure to trigger
indentation intelligently, such as after typing "Host " in the beginning
of a line.  Unfortunately, don't handle other forms of whitespace in
that last example.

jaki committed:
refactor: use 0=<word> instead of =<word>

Not sure if it makes any practical difference, but since others do it,
might as well do it.

jaki committed:
test: add sshconfig indent test

...that currently fails.

jaki committed:
test: add corresponding sshconfig.ok

jaki committed:
fix: indent non-empty lines

The indent for lines under "Host" or "Match" lines only works for empty
lines.  This is sufficient during editing, but applying the indentation
globally on a file would not pick this up and result in no indentation.
Fix this by removing the empty line restriction.

jaki committed:
chore: set Bram as maintainer, Jason as author

jaki committed:
refactor: let prev_lnum, curr_lnum

jaki committed:
fix: set default indent to prev line's indent

Set the default indent level to the previous line's indent level.  The
`sshconfig` indent test should now pass.

k-takata commented on runtime/indent/sshconfig.vim:
```diff
@@ -0,0 +1,35 @@
+" Vim indent file
+" Language: ssh config file
+" Maintainer: Bram Moolenaar <Bram@vim.org>
+" Original Author: Jason Kim <git@jasonk.me>
```
You are the maintainer, aren't you?

jaki committed:
chore: set Jason as maintainer

jaki committed:
style: remove original author

...because just having maintainer is fie.  Also, change the last
modified date format to "%Y %b %d".

brammool commented:
I'll include it, thanks.  Don't forget to send updates when you make improvements.
